### PR TITLE
feat: bootstrap azure module

### PR DESCRIPTION
**Pull request by fireballpoint1**
Bootstrap module for azure. Created app registration, service principle, storage account - container - blob storage and resource groups. 
Implemented output file as well.

### CODE REVIEW

**Comment by krtk6160** [comment]
Please follow the conventional commits guidelines linked in [DEV.md](https://github.com/GaloyMoney/galoy/blob/main/DEV.md#contributing) for commit messages and PR titles

File: modules/bootstrap/azure/variables.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+variable "name_prefix" {}
+variable "external_users" { default = [] }
```
Comment (sandipndev) [inline modules/bootstrap/azure/variables.tf:2]: Can you get rid of `external_users`, it doesn't look like it is needed here.

File: modules/bootstrap/azure/tf-state-storage.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+resource "azurerm_storage_account" "bootstrap" {
+  name                     = "${local.name_prefix}sa"
```
Comment (sandipndev) [inline modules/bootstrap/azure/tf-state-storage.tf:2]: any account name must be exported (added to outputs)

File: modules/bootstrap/azure/resource-group.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Create resource group
+resource "azurerm_resource_group" "bootstrap" {
+  name     = "${local.name_prefix}-bootstrap"
+  location = local.resource_group_location
+}
```
Comment (sandipndev) [inline modules/bootstrap/azure/resource-group.tf:5]: please add a newline at the end. (EOLF)

File: modules/bootstrap/azure/service-principal.tf [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Configure the Azure Active Directory Provider
+provider "azuread" {
+  tenant_id = local.tenant_id
+}
+
+locals {
+  inception_app_name = "${local.name_prefix}-inception-tf"
```
Comment (sandipndev) [inline modules/bootstrap/azure/service-principal.tf:7]: let's keep all the locals along with variables.tf

**Review by sandipndev — CHANGES_REQUESTED** [review]
I have tested the resources in this PR and both the apply and destroy phase worked! 
Please make the formatting changes and this PR should be ready to go in.

**Review by sandipndev — APPROVED** [review]

**Pull request merged**